test-it/karma/javascriptinvoke/CloseReference.Test.js

```diff
@@ -0,0 +1,146 @@
+            expect(viPathParser('errorInAndInvalidReferenceError.status')).toBeTrue();
+            expect(viPathParser('errorInAndInvalidReferenceError.code')).toBe(100);
+            expect(viPathParser('errorInAndInvalidReferenceError.source')).toBe('error');
+            expect(viPathParser('isNotAValidRefnum4')).toBeTrue();
+            done();
+        });
+    });
+
+    it('does not close a JavaScript Static Reference', function (done) {
+        var viName = 'StaticReferencesVI';
+        var runSlicesAsync = vireoRunner.rebootAndLoadVia(vireo, jsCloseReferenceViaUrl);
+        var viPathParser = vireoRunner.createVIPathParser(vireo, viName);
+        vireoRunner.enqueueVI(vireo, viName);
+
+        runSlicesAsync(function (rawPrint, rawPrintError) {
+            expect(rawPrint).toBeEmptyString();
+            expect(rawPrintError).toBeEmptyString();
+            expect(viPathParser('getObjectReferenceError.status')).toBeFalse();
+            expect(viPathParser('getObjectReferenceError.code')).toBe(0);
+            expect(viPathParser('getObjectReferenceError.source')).toBeEmptyString();
+            expect(viPathParser('noErrorInAndValidReferenceError.status')).toBeFalse();
+            expect(viPathParser('noErrorInAndValidReferenceError.code')).toBe(0);
+            expect(viPathParser('noErrorInAndValidReferenceError.source')).toBeEmptyString();
+            expect(viPathParser('isNotAValidRefnum1')).toBeFalse();
+            expect(viPathParser('errorInAndValidReferenceError.status')).toBeTrue();
+            expect(viPathParser('errorInAndValidReferenceError.code')).toBe(100);
+            expect(viPathParser('errorInAndValidReferenceError.source')).toBe('error');
+            expect(viPathParser('isNotAValidRefnum2')).toBeFalse();
+            done();
+        });
+    });
+
+    it('makes a JavaScript Dynamic Reference invalid for further use', function (done) {
+        var viName = 'InvalidReferenceParameterVI';
+        var runSlicesAsync = vireoRunner.rebootAndLoadVia(vireo, jsCloseReferenceViaUrl);
+        var viPathParser = vireoRunner.createVIPathParser(vireo, viName);
+        vireoRunner.enqueueVI(vireo, viName);
+
+        runSlicesAsync(function (rawPrint, rawPrintError) {
+            expect(rawPrint).toBeEmptyString();
+            expect(rawPrintError).toBeEmptyString();
+            expect(viPathParser('getObjectReferenceError.status')).toBeFalse();
+            expect(viPathParser('getObjectReferenceError.code')).toBe(0);
+            expect(viPathParser('getObjectReferenceError.source')).toBeEmptyString();
+            expect(viPathParser('closeReferenceError.status')).toBeFalse();
+            expect(viPathParser('closeReferenceError.code')).toBe(0);
+            expect(viPathParser('closeReferenceError.source')).toBeEmptyString();
+            expect(viPathParser('isNotAValidRefnum1')).toBeTrue();
+            expect(viPathParser('invalidReferenceError.status')).toBeTrue();
```

Comment:
invalidReferenceError is likely the only thing this test really needs to assert. Do we want all the previous ones? If you think it is safe to keep those, then put them in a separate pre-condition function so that it is clear what the test is intending to assert (using expect statements here) and what the preconditions are. This feedback is applicable for the other test methods too.

Reply:
The tests are not longer karma based. They are now vtr based and this problem is not as bad.